docs/managedCodeGen.md

```diff
@@ -0,0 +1,29 @@
+# Code generation for managed react-native modules
+
+This is the feature is applicable to C# implemented modules or viewmanagers. It will make sure that the code needed to integrate with the react-native application is constructed at build time instead of at runtime.
+
+In earlier versions of react-native-windows managed modules registered themselves based on reflection at runtime. This method has a few disadvantages. The error handling was limmited and only availalbe at runtime. The cost for analyzing was paid each and every time the app ran.
+Doing it at build time should result in developers seeing all errors up front, rather than only seeing errors for the code being tested. And the runtime cost is not incurred every single time.
+
+## How it works
+This change adds a new targets file that is imported from the [CSharpLib.targets](https://github.com/microsoft/react-native-windows/blob/master/vnext/PropertySheets/External/Microsoft.ReactNative.Uwp.CppLib.targets) and [CSharpApp.targets](https://github.com/microsoft/react-native-windows/blob/master/vnext/PropertySheets/External/Microsoft.ReactNative.Uwp.CppApp.targets). That will enable the codegen for managed projects.
+
+### In source mode
+The targets file hooks up an msbuild target that will ensure the tool is deployed and runnable. The codegen tool is defined in the Microsoft.ReactNative.Managed.CodeGen project. It will build and deploy the tool in the current version of your project (except for ARM and Arm64 it will use x86 and x64). It will then collect all the C# source files, assembly references and defines from the app (or library) and invoke the tool to generate a C# file. This is very similar to how Xaml codegen works.
```

Comment:
I think the tool itself should always be x64, we don't have to support x86 (and certainly don't have to support ARM and ARM64) **developer** environments.

Reply:
yea, I changed that in the PR, but missed it here.